include/nbl/builtin/hlsl/glsl_compat/math.hlsl

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_GLSL_COMPAT_CORE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_GLSL_COMPAT_CORE_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/spirv_intrinsics/core.hlsl"
+#include "nbl/builtin/hlsl/type_traits.hlsl"
+
+#ifndef __HLSL_VERSION
+#include <vector3d.h>
+#endif
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace glsl
+{
+
+template<typename T>
+inline T radians(NBL_CONST_REF_ARG(T) degrees)
+{
+	static_assert(
+		is_floating_point<T>::value,
+		"This code expects the type to be either a double or a float."
+	);
+
+	return degrees * PI<T>() / T(180);
+}
+
+template<typename T>
+inline T degrees(NBL_CONST_REF_ARG(T) radians)
+{
+	static_assert(
+		is_floating_point<T>::value,
+		"This code expects the type to be either a double or a float."
+	);
+
+	return radians * T(180) / PI<T>();
+}
```

Comment:
we have `numbers` ot something like that for PI and friends.

Also the `_HLSL_VERSION` case should forward to the inline SPIR-V from the std450 extended set